src/main/scala/scala/xml/parsing/MarkupParserCommon.scala

```diff
@@ -250,7 +250,6 @@ private[scala] trait MarkupParserCommon extends TokenTests {
         else if (ch == SU || eof)
           truncatedError("") // throws TruncatedXMLControl in compiler
 
-        sb append ch
```

Comment:
This is the line in scala-xml that in turn causes the OOM exception to be thrown by `java.lang.StringBuilder` by way of `java.util.Arrays`.  Even after removing the line in question from scala-xml, it still causes the test suite to run out of memory.  Which scala-xml is it running that causes the memory overflow?  It doesn't seem to be the source one...